docs/user-namespaces.md

```diff
@@ -0,0 +1,102 @@
+
+First check your containerd, Linux and Kubernetes versions. If those are okay, then there is no
+special configuration needed on conntainerd. You can just follow the steps in the [Kubernetes
+website][kube-example].
+
+[kube-example]: https://kubernetes.io/docs/tasks/configure-pod-container/user-namespaces/
+
+# Limitations
+
+You can check the limitations Kubernetes has [here][kube-limitations]. Note that different
+Kubernetes versions have different limitations, be sure to check the site for the Kubernetes version
+you are using.
+
+Different containerd versions have different limitations too, those are highlighted in this section.
+
+[kube-limitations]: https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#limitations
+
+### containerd 1.7
+
+One limitation present in containerd 1.7 is that it needs to change the ownership of every file and
+directory inside the container image, during Pod startup. This means it has a storage overhead (the
+size of the container image is duplicated each time a pod is created) and can significantly impact
+the container startup latency.
+
+You can mitigate this limitation by switching `/sys/module/overlay/parameters/metacopy` to `Y`. This
+will significantly reduce the storage and performance overhead, as only the inode for each file of
+the container image will be duplicated, but not the content of the file. This means it will use less
+storage and it will be faster. However, it is not a panacea.
+
+If you change the metacopy param, make sure to do it in a way that is persistant across reboots. You
+should also be aware that this setting will be used for all containers, not just containers with
+user namespaces enabled. This will affect all the snapshots that you take manually (if you happen to
+do that). In that case, make sure to use the same value of `/sys/module/overlay/parameters/metacopy`
+when creating and restoring the snapshot.
+
+### containerd 2.0
+
+The storage and latency limitation from containerd 1.7 are not present in container 2.0 and above,
+if you use the overlay snapshotter (this is used by default). It will not use more storage at all,
+and there is no startup latency.
+
+This is achieved by using the kernel feature idmap mounts with the container rootfs (the container
+image). This allows an overlay file-system to expose the image with different UID/GId without copying
+the files nor the inodes, just using a bind-mount.
+
+You can check if you are using idmap mounts for the container image if you create a pod with user
+namespaces, exec into it and run:
+
+```
+mount | grep overlay
+```
+
+You should see a reference to the idmap mount in the `lowerdir` parameter, in this case we can see
+`idmapped` used there:
+
+```
+overlay on / type overlay (rw,relatime,lowerdir=/tmp/ovl-idmapped823885363/0,upperdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/1018/fs,workdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/1018/work)
+```
```

Comment:
add your sample w/success here / cut paste a snip of the config.json important parts enabling userns

Reply:
The config.json was created by Kubernete following the linked example

Reply:
Added a config.json to use with ctr, I think that should be covered now! :)